### PR TITLE
Add contributor guide for docs and fix links

### PR DESCRIPTION
**Pull request by willingc**
This PR has a few minor fixes to external links. It also adds a first pass at a Contributor Guide for Documentation.

Let me know if you want me to split the PR into 2 PRs (one with minor fixes and another with the guide).

### CODE REVIEW

File: docs/source/migrating.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@
 Migrating from IPython
 ======================
 
-`The Big Split <https://blog.jupyter.org/2015/04/15/the-big-split/>`__
+`The Big Split <http://blog.jupyter.org/2015/04/15/the-big-split/>`__
```
Comment (minrk) [inline docs/source/migrating.rst]: This should stay https

File: docs/source/subprojects.rst [hunk 2]
```diff
@@ -51,7 +51,7 @@ powerful environment for interactive computing in Python.
 
     `IPython <http://ipython.org>`__
         interactive computing in Python.
-        `Documentation <https://ipython.org/documentation.html>`__ |
+        `Documentation <http://ipython.org/documentation.html>`__ |
```
Comment (minrk) [inline docs/source/subprojects.rst]: This should stay https, as well.

File: docs/source/migrating.rst (hunk 1 above)
Comment (willingc) [inline docs/source/migrating.rst]: @minrk Will do. Oddly, ``make linkcheck`` complains about the SSL. I will change the docs back to https and try to figure out why ``make linkcheck`` errors.

Comment (minrk) [inline docs/source/migrating.rst]: I think certain versions of Python on certain platforms actually have a non-functional SSL library. What version of Python are you using on what platform?

Comment (willingc) [inline docs/source/migrating.rst]: @minrk I suspect that you are correct that it is an OpenSSL issue with OS X. 
Python 3.5
OpenSSL 0.9.8zg 14 July 2015
OS X 10.10.5

**Pull request merged**